### PR TITLE
Example Code Cleanup

### PR DESCRIPTION
**Pull request by JonoAugustine**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing
[pull-request]: https://github.com/JDA-Applications/JDA-Utilities/pulls

#### Pull Request Checklist
Please follow the following steps before opening this PR.<br>
PRs that do not complete the checklist will be subject to denial for
missing information.

- [X] I have checked the [pull request page][pull-request] for upcoming
      or merged features/bug fixes.
- [X] I have read JDA's [contributing guidelines][contributing].

#### Pull Request Information
Check and fill in the blanks for all that apply:

- [ ] My PR fixes a bug, error, or other issue with the library's codebase.
- [X] My PR is for the `Example` module of the JDA-Utilities library.
- [ ] My PR creates a new module for the JDA-Utilities library: `______`.

#### Description

I have improved the readability of the About and GuildList example commands so that newcomers can get a better grasp on how it all works.

Both
- Max line ~80-90 char
- Spaced out chained ``.append()``  calls
Guild List
- Added Bot-Human % to each guild
- Paginates w/ numbered items
About
- Extracted sending code to ``send(event)`` and placed inside of ``getApplicationInfo.queue`` for failure and success responses so building the InvLink doesn't use ``complete()``
- Changed ``features`` param to ``Command[] commands``
   - Makes setup simpler
- Added 2 more constructors, 1 which takes a pre-made Invite link instead of a permission array

### CODE REVIEW

**Comment by jagrosh** [comment]
Right off the bat, please don't change the copyright, and please don't remove the original author from the file if you are modifying it.

**Comment by JonoAugustine** [comment]
woopsies, force of habit. ill change that

**Comment by MinnDevelopment** [comment]
It would be extremely helpful to include some kind of description or explanation as to what these changes exactly improve/change.

**Comment by jagrosh** [comment]
I also noticed you included a `sun` import, which is probably not very good practice considering some people might be building on OpenJDK or future releases. There is a JDA util called `Checks` that might have similar functionality and is already a dependency.

**Comment by JonoAugustine** [comment]
> I also noticed you included a `sun` import, which is probably not very good practice considering some people might be building on OpenJDK or future releases. There is a JDA util called `Checks` that might have similar functionality and is already a dependency.

Thanks, was unaware of this

**Comment by JonoAugustine** [comment]
@MinnDevelopment updated description

File: examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java [hunk 1]
```diff
@@ -15,107 +15,205 @@
...
+{
+
     private final Color color;
     private final String description;
-    private final Permission[] perms;
-    private String oauthLink;
-    private final String[] features;
-    
-    public AboutCommand(Color color, String description, String[] features, Permission... perms)
+    private final Command[] commands;
```
Comment (jagrosh) [inline examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java:51]: This shouldn't be stored here, as `event.getClient().getCommands()` will provide the currently-available command list.

File: examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java [hunk 2]
```diff
@@ -15,107 +15,205 @@
...
+     * The default constructor, this is used to clean the clutter from other constructors
+     *
+     * @param color The Embed Color
+     * @param description The bot's Description
+     * @param commands All the commands to show
+     */
+    private AboutCommand(Color color, String description, Command[] commands)
+    {
+        //Super variables
+        this.name = "About";
```
Comment (jagrosh) [inline examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java:103]: This should be all lowercase `about`

File: examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java [hunk 3]
```diff
@@ -15,107 +15,205 @@
...
+        EmbedBuilder builder = new EmbedBuilder()
+            .setColor(color)
+            .setAuthor(self.getName(), null, self.getAvatarUrl())
+            .setTitle("About " + self.getName() + "!")
+            .setFooter("Last restart", self.getAvatarUrl())
+            .setTimestamp(cmdClient.getStartTime());
+
+        //Use a StringBuilder cuz nice
+        sb.append("Hello! I am **").append(self.getName()).append("**\n")
+          .append("I was made by **").append(ownerMention)
```
Comment (jagrosh) [inline examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java:156]: I'm not sure why you removed the ability to change whether or not the owner is the author; this is used in self-hostable bots where the owner and the author are likely different people,

File: examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java [hunk 4]
```diff
@@ -15,107 +15,205 @@
...
+
+        //Setup the basic parts of the EmbedBuilder
+        EmbedBuilder builder = new EmbedBuilder()
+            .setColor(color)
+            .setAuthor(self.getName(), null, self.getAvatarUrl())
+            .setTitle("About " + self.getName() + "!")
+            .setFooter("Last restart", self.getAvatarUrl())
+            .setTimestamp(cmdClient.getStartTime());
+
+        //Use a StringBuilder cuz nice
```
Comment (jagrosh) [inline examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java:154]: 'cuz nice' doesn't feel like documentation improvement here

File: examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java [hunk 5]
```diff
@@ -15,107 +15,205 @@
...
+        for (Command cmd: commands)
         {
-            builder.addField("Stats", (event.getClient()).getTotalGuilds() + " Servers\nShard " + (event.getJDA().getShardInfo().getShardId() + 1) 
-                    + "/" + event.getJDA().getShardInfo().getShardTotal(), true);
-            builder.addField("This shard", event.getJDA().getUsers().size() + " Users\n" + event.getJDA().getGuilds().size() + " Servers", true);
-            builder.addField("", event.getJDA().getTextChannels().size() + " Text Channels\n" + event.getJDA().getVoiceChannels().size() + " Voice Channels", true);
+            if (cmd.isOwnerCommand()) continue;
+            //You may need to use a Menu Paginator if there are too many Commands
+            //  for 1 MessageEmbed
+            sb.append(cmd.getName()).append("\n");
```
Comment (jagrosh) [inline examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java:183]: This should definitely not be the list of commands, many bots have a huge number of commands and definitely don't want them listed here since they are available in the help. This should remain a list of developer-defined features.

File: examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java [hunk 6]
```diff
@@ -15,107 +15,205 @@
...
-    
-    public void setIsAuthor(boolean value)
+
+
+    @Override
+    protected void execute(CommandEvent event)
     {
-        this.IS_AUTHOR = value;
+        //Get info about the bot
+        event.getJDA().asBot().getApplicationInfo().queue(info -> {
```
Comment (jagrosh) [inline examples/src/main/java/com/jagrosh/jdautilities/examples/command/AboutCommand.java:120]: Using `queue` here is rather unnecessary, as the version with `complete` only calls it once: the very first time the command is run. We also don't want to fetch this info every time the command is run, since it rarely changes.

File: gradle.properties [hunk 7]
```diff
@@ -1,18 +0,0 @@
-#
```
Comment (jagrosh) [inline gradle.properties:1]: Why was this file deleted?

File: examples/src/main/java/com/jagrosh/jdautilities/examples/command/GuildlistCommand.java [hunk 8]
```diff
@@ -47,35 +53,80 @@
...
+            });
+    }
+
+    /**
+     * @param guild The guild to count
+     * @return the number of members which are not bots
+     */
+    private static int getHumanCount(Guild guild)
+    {
+        return guild.getMembers().size() - guild.getMembers().stream()
```
Comment (jagrosh) [inline examples/src/main/java/com/jagrosh/jdautilities/examples/command/GuildlistCommand.java:99]: Why convert to an array instead of `guild.getMembers().stream().filter(m -> m.getUser().isBot()).count()`?

**Review by jagrosh — CHANGES_REQUESTED** [review]
Overall, it feels like you're completely changing the feel of the AboutCommand. It's true that the code could be cleaned up, but the output should remain nearly identical.
